### PR TITLE
feat: add glyph ref icons

### PR DESCRIPTION
**Pull request by mortada-codes**
This PR...

## Changes

- add glyph icons for incoming and outgoing refs

## Fixes

-

## How to test it

- select a resource from the navigator pane with incoming or outgoing links
- should see the relation icon in the editor source pane

## Screenshots

- 
Screen Shot 2022-01-13 at 2 31 05 PM

https://user-images.githubusercontent.com/20525304/149487961-529dc150-e84d-47a8-bdc9-3762c400f3a4.mp4

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by olensmar** [comment]
nice work @mortada-codes! could we add a popup on these glyphs showing the incoming refs for that line (there could be multiple) - clickable so that clicking one would take you to the "source" of the ref?

**Comment by olensmar** [comment]
thanks @mortada-codes - looking good - two comments:
- formatting needs to be like in the popups in the navigator - i.e. 
image
instead of 
image

- the hover seems to happen when hovering to the left of the icon - not on top of it

**Comment by mortada-codes** [comment]
I updated `react-monaco-editor` `monaco-editor` `monaco-editor-webpack-plugin`
then updated the webpack configuration  but i faced the same error like this issue https://github.com/microsoft/monaco-editor/issues/2893 
should we hold glyph hover message till fix? or we can extend the css `hover-contents` to make it nicer but not the optimal solution

**Comment by olensmar** [comment]
thanks @mortada-codes - then we can leave it as is for now - it's still a great addition!

File: src/components/molecules/Monaco/codeIntel.ts [hunk 1]
```diff
@@ -142,75 +142,59 @@ export async function applyForResource(
...
-      hasUnsatisfiedRef ? InlineDecorationTypes.UnsatisfiedRef : InlineDecorationTypes.SatisfiedRef
-    );
-    newDecorations.push(inlineDecoration);
-
     const commandMarkdownLinkList: monaco.IMarkdownString[] = [];
-    outgoingRefs.forEach(outgoingRef => {
-      if (!outgoingRef.target) {
+    matchedRefs.forEach(matchRef => {
+      if (!matchRef.target) {
         return;
```
[the remark this answers is not in the document]
Comment (devcatalin) [reply src/components/molecules/Monaco/codeIntel.ts:178]: it is, in this case it just skips over that specific item and goes to the next, similar to a `break` in a traditional for loop, but it doesn't return anything

**Review by devcatalin — APPROVED** [review]
LGTM

**Pull request merged**